.s2i/bin/assemble

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+ASSEMBLE_SCRIPT="/usr/libexec/s2i/assemble"
+
+pushd /tmp/src || exit 1
+
+echo "---> Building your Ruby application from source ..."
+if [ -f Gemfile ]; then
+  ADDTL_BUNDLE_ARGS="--retry 2"
+  if [ -f Gemfile.lock ]; then
+    # This is new in version 3.0
+    # [DEPRECATED] The `--deployment` flag is deprecated because it relies
+    # on being remembered across bundler invocations,
+    # which bundler will no longer do in future versions.
+    # Instead please use `bundle config set --local deployment 'true'`,
+    # and stop using this flag
+    bundle config set --local deployment 'true'
+  fi
+
+  if [[ "$RAILS_ENV" == "development" || "$RACK_ENV" == "development" ]]; then
+    BUNDLE_WITHOUT=${BUNDLE_WITHOUT:-"test"}
+  elif [[ "$RAILS_ENV" == "test" || "$RACK_ENV" == "test" ]]; then
+    BUNDLE_WITHOUT=${BUNDLE_WITHOUT:-"development"}
+  else
+    BUNDLE_WITHOUT=${BUNDLE_WITHOUT:-"development:test"}
+  fi
+
+  if [ -n "$BUNDLE_WITHOUT" ]; then
+    # [DEPRECATED] The `--without` flag is deprecated because
+    # it relies on being remembered across bundler invocations,
+    # which bundler will no longer do in future versions.
+    # Instead please use `bundle config set --local without ''test''`,
+    # and stop using this flag
+    bundle config set --local without '$BUNDLE_WITHOUT'
+  fi
+
+  echo "---> Running 'bundle install ${ADDTL_BUNDLE_ARGS}' ..."
+  # [DEPRECATED] The `--path` flag is deprecated because
+  # it relies on being remembered across bundler invocations,
+  # which bundler will no longer do in future versions.
+  # Instead please use `bundle config set --local path './bundle'`,
+  # and stop using this flag
+  bundle config set --local path './bundle'
+  bundle install ${ADDTL_BUNDLE_ARGS}
+
+  echo "---> Cleaning up unused ruby gems ..."
+  bundle clean --verbose
+fi
+
+if [[ "$RAILS_MASTER_KEY" == "" && ! $(ls config/master.key 2>/dev/null 1>/dev/null) ]]; then
+	if [[ $(ls config/secrets.yml.enc 2>/dev/null 1>/dev/null) ]]; then
```

Comment:
Wouldn't it just work, without this error?

Like, it might, and regenerate/ignore the file?

Reply:
Likewise, if it's needed, it's better to check with `-e`.

Reply:
It would not regenerate it (it can't get to the original data) and without the key, it was encrypted with, it becomes undecryptable, and `rake assets:precompile` will fail in that case on some OpenSSL exception.

What could be done is to delete the file so it is properly encrypted with the new key, but any contents are lost either way.

Reply:
Good to know!

Reply:
TBH I'd rather delete it in that case just to "recover".. and maybe throw some warning?

Reply:
Throwing a warning sounds good.